packages/grid/_modules_/grid/constants/eventsConstants.ts

```diff
@@ -1,85 +1,431 @@
+
+/**
+ * Fired while an element or text selection is dragged over the column header.
+ * It's mapped to the `dragover` DOM event.
+ * Called with a [[GridColumnHeaderParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_HEADER_DRAG_OVER = 'columnHeaderDragOver';
+
+/**
+ * Fired when the dragged column header enters a valid drop target.
+ * It's mapped to the `dragend` DOM event.
+ * Called with a [[GridColumnHeaderParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_HEADER_DRAG_ENTER = 'columnHeaderDragEnter';
+
+/**
+ * Fired when the dragging of a column header ends. Called with a [[GridColumnHeaderParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_HEADER_DRAG_END = 'columnHeaderDragEnd';
 
+/**
+ * Fired when the selection state of one or multiple rows changes.
+ * Called with a [[GridSelectionModelChangeParams]] object.
+ * @event
+ */
 export const GRID_SELECTION_CHANGED = 'selectionChange';
 
+/**
+ * Fired when the current page has changed. Called with a [[GridPageChangeParams]] object.
+ * @event
+ */
 export const GRID_PAGE_CHANGED = 'pageChange';
+
+/**
+ * Fired when the page size has changed. Called with a [[GridPageChangeParams]] object.
+ * @event
+ */
 export const GRID_PAGESIZE_CHANGED = 'pageSizeChange';
 
-export const GRID_ROWS_SCROLL = 'scrolling:rows';
-export const GRID_ROWS_SCROLL_END = 'scroll:rowEnd';
+/**
+ * Fired during the scroll of the grid viewport. Called with a [[GridScrollParams]] object.
+ * @event
+ */
+export const GRID_ROWS_SCROLL = 'rowsScroll';
+
+/**
+ * Fired when scrolling to the bottom of the grid viewport. Called with a [[GridRowScrollEndParams]] object.
+ * @event
+ */
+export const GRID_ROWS_SCROLL_END = 'rowsScrollEnd';
 
+/**
+ * Fired when a `mousedown` DOM event happens in the column header separator.
+ * Called with a [[GridColumnHeaderParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_SEPARATOR_MOUSE_DOWN = 'columnSeparatorMouseDown';
+
+/**
+ * Fired during the resizing of a column. Called with a [[GridColumnResizeParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_RESIZE = 'columnResize';
+
+/**
+ * Fired when a column is resized. Called with a [[GridColumnResizeParams]] object.
+ * @event
+ */
 export const GRID_COLUMN_RESIZE_COMMITED = 'columnResizeCommited';
+
+/**
+ * Fired when the user starts resizing a column. Called with an object `{ field: string }`.
+ * @event
+ */
 export const GRID_COLUMN_RESIZE_START = 'columnResizeStart';
+
+/**
+ * Fired when the user starts resizing a column. Called with an object `{ field: string }`.
+ * @event
+ */
 export const GRID_COLUMN_RESIZE_STOP = 'columnResizeStop';
 
+/**
+ * Fired when the user ends resizing a column.
+ * @event
+ */
 export const GRID_COLUMN_ORDER_CHANGE = 'columnOrderChange';
 
+/**
+ * Fired when the rows are updated.
+ * @event
+ */
 export const GRID_ROWS_UPDATED = 'rowsUpdated';
+
+/**
+ * Fired when the rows are updated.
+ * @event
+ */
 export const GRID_ROWS_SET = 'rowsSet';
```

Comment:
I couldn't understand the difference between this one and `GRID_ROWS_UPDATED`.

Reply:
`GRID_ROWS_SET` is published when the full set of rows is reset.
`GRID_ROWS_UPDATED ` is published when you update one or more rows.

Reply:
I think that the underlying question is: why does the difference matter (from a developer problem perspective)? For instance, the data grid handles the two events the same way. What if we merged them?

Reply:
No it should not be the same. One reset the prop and do a full rerender and the other one is more optimised

Reply:
@dtassone Can you describe the difference in a short text to add in the description?

Reply:
`GRID_ROWS_SET` Fired when the rows are reset
`GRID_ROWS_UPDATED` Fired when one or multiple rows received a delta update

Reply:
@dtassone I mean, Why does the difference matter? Currently, the data grid listens to the two and performs the same action. So it looks like these developers need to do the same, listens to the two events. If they don't, they will get surprises.

> Fired when the rows are reset

How will a developer understand what "reset" mean?

> Fired when one or multiple rows received a delta update

Would it be clear if we explicitly mention the imperative apiRef methods that allow performing a delta update?

Reply:
> Would it be clear if we explicitly mention the imperative apiRef methods that allow performing a delta update?

Why not.

> Should we mention that it triggers when the rows prop change?

Not only that, it's also when rows length change, and it's doing a sort of internal reset of rows.

> How will a developer understand what "reset" mean?

Reset word is pretty common, in the computer world...
How should we name it?

Reply:
> it's also when rows length change

Oh so if a developer uses the imperative API to add or remove records, then `GRID_ROWS_SET` is fired. I thought it was only for the `rows` prop. (One example of confusion in the description 😁). Maybe we could say:

- GRID_ROWS_SET: Fired when the length of the rows changes.
- GRID_ROWS_UPDATED: Fired when at least one row is updated in place.

Reply:
I would not document both `GRID_ROWS_UPDATED` and `GRID_ROWS_SET`. It's not clear the conditions in which they are fired. 

> How will a developer understand what "reset" mean? Should we mention that it triggers when the rows prop change?

It's not triggered when the `rows` prop change, but when calling `apiRef.current.updateRows`. This should be clear in the description.

> Reset word is pretty common, in the computer world...

IHMO this "reset" word may confuse users. One of the [definitions](https://www.dictionary.com/browse/reset) is "to set again". But in the React world "set again" may not cause a rerender if the value is the same. If we are using "reset" in the context of when the value of something changes, this should be more explicit, maybe using "change".

Reply:
Feel free to call it `GRID_ROWS_CHANGE` I have no strong opinion anymore.

Reset as it does a clear and set to update the sort, filters...

>GRID_ROWS_SET: Fired when the length of the rows changes or when a new set of rows is provided.
GRID_ROWS_UPDATED: Fired when at least one row is updated in place.

👍 


I don't see why we are restricting, if some advanced trading app needs to use those events, let them do, that's why the grid is based on an event emitter, so it could be hacked, customized... 
We can reassess at a later stage, this code will be refactored along with time, things will evolve over the versions.